docs/_chapters/230-manifest-annotations.md

```diff
@@ -1,136 +1,284 @@
+
+@ASL_2_0
+class Foo { ... }
+```
+
+### Adding Attributes and Directives
+
+When creating *custom bundle annotations* a common requirement is to make them parameterizable such that the values of the *custom bundle annotation* feed into the header clauses resulting from the *bundle annotation* applied to it (*remember; a **custom bundle annotation** is meta-annotated with a **bundle annotation***.)
+
+[OSGi](https://osgi.org/specification/osgi.core/7.0.0/) specifies two annotations, [`@Attribute`](https://osgi.org/specification/osgi.core/7.0.0/framework.api.html#org.osgi.annotation.bundle.Attribute) and [`@Directive`](https://osgi.org/specification/osgi.core/7.0.0/framework.api.html#org.osgi.annotation.bundle.Directive), for this purpose. Any methods of the **custom bundle annotation** annotated with `@Attribute` or  `@Directive` result in those becoming additional attributes or directives respectively of the resulting header clause.
+
+##### `@Attribute`
+
+`@Attribute` allows you to add new or update existing attributes from the _bundle annotation_.
+
+```java
+@Capability(namespace = "foo.namespace")
+@interface Extended {
+	@Attribute("foo.attribute") // this attribute enhances the @Capability
+	String value() default "bar";
+}
+
+// usage
+
+@Extended
+class Foo {}
+```
+
+which results in the manifest header:
+
+`Provide-Capability: foo.namespace;foo.attribute=bar`
+
+##### `@Directive`
+
+`@Directive` behaves similarly; with some caveats. You can add new or update existing directives for _non-specified namespaces_.
+
+```java
+@Capability(namespace = "foo.namespace")
+@interface Extended {
+	@Directive("foo.directive")
+	String value() default "bar";
+}
+
+// usage
+
+@Extended
+class Foo {}
+```
+
+results in the manifest header:
+
+`Provide-Capability: foo.namespace;foo.directive:=bar`
+
+**However**, _specified namespaces_ will be validated and will not accept directives which are not part of the spec _unless_ they are prefixed with `x-`.
+
+```java
+@Capability(namespace = "osgi.extender", name = "bar", version = "1.0.0")
+@interface Extended {
+	@Directive("foo")
+	String value() default "bar";
+}
+
+// usage
+
+@Extended
+public class Foo {}
+```
+
+will result in an error:
+
+`Unknown directive 'foo:' for namespace 'osgi.extender' in 'Provide-Capability'. Allowed directives are [effective:,uses:], and 'x-*'.`
+
+This next example however:
+
+```java
+@Capability(namespace = "osgi.extender", name = "bar", version = "1.0.0")
+@interface Extended {
+	@Directive("x-foo")
+	String value() default "bar";
+}
+
+// usage
+
+@Extended
+public class Foo {}
+```
+
+results in the manifest header:
+
+`Provide-Capability: osgi.extender;osgi.extender=bar;version:Version="1.0.0";x-foo:=bar`
+
+### Where to find Bundle Annotations
+
+**OSGi** *bundle annotations* can be found in the `osgi.annotation` bundle.
+
+```xml
+<dependency>
+  <groupId>org.osgi</groupId>
+  <artifactId>org.osgi.annotation.bundle</artifactId>
```

Comment:
This isn't the maven artifact for `osgi.annotation` is it meant to be?

Reply:
hmm, I'm always torn with the various annotation bundles. In general I feel that using the most specific dependency is best. Other times I use the `osgi.annotation` aggregate out of convenience. I'll follow consensus on this though.

Reply:
> In general I feel that using the most specific dependency is best.

I'm fine with that approach, it's just that line 213 explicitly states `osgi.annotation`. Whichever we use these two places should be consistent.